qiskit/synthesis/discrete_basis/solovay_kitaev.py

```diff
@@ -91,7 +91,7 @@ def run(
         recursion_degree: int,
         return_dag: bool = False,
         check_input: bool = True,
-    ) -> "QuantumCircuit" | "DAGCircuit":
+    ) -> "QuantumCircuit" | "DAGCircuit":  # noqa: F821
```

Comment:
If the correct fix here is to disable F821, then I think we should probably have it globally disabled - this kind of pattern is common, so the lint rule is getting in the way.

Reply:
F821 added to `ignore` list in 80b078b967c03d9a64f50accae1c166adf506e91